lib/puppet/parser/resource.rb

```diff
@@ -199,6 +199,51 @@ def raw_tagged?(tag_array)
     super || ((scope_resource = scope.resource) && !scope_resource.equal?(self) && scope_resource.raw_tagged?(tag_array))
   end
 
+  # Fill resource params from a capability
+  #
+  # This backs 'consumes => Sql[one]'
```

Comment:
We should put `@api private` on methods we do not want people to be calling from their own logic.